src/main/java/TaskList.java

```diff
@@ -0,0 +1,290 @@
+                    if (storingList.get(numDone - 1).isDone) {
+                        //if already true
+                        throw new GrapieExceptions(ErrorMsg.taskNumIsAlreadyDone(numDone));
+                    } else {
+
+                        storingList.get(numDone - 1).isDone = true;
+                        //storingList.set(taskNum - 1, updatedTask);
+
+                        String printStr =
+                                "Nice! I've marked this task as done: \n" + storingList.get(numDone - 1);
+                        //formattingDivider(printStr);
+                        storage.replaceLineFromHardDisk(numDone);
+
+                        return printStr;
+
+                    }
+                } else {
+                    throw new GrapieExceptions(ErrorMsg.numberDoNotExistError(numDone));
+                }
+            } else {
+                throw new GrapieExceptions(ErrorMsg.invalidNumberError);
+            }
+        }
+    }
+
+    /**
+     * Delete task the user specifies.
+     * Help check if the task number exists.
+     *
+     * @param inputStr User's input, delete command and task to be deleted.
+     * @throws GrapieExceptions Error if the task number to be deleted do not exist.
+     * @throws IOException      Throws away the exception.
+     */
+    public String deleteTask(String inputStr) throws GrapieExceptions, IOException {
+        if (inputStr.length() <= 7) {
+            throw new GrapieExceptions(ErrorMsg.invalidNumberError);
+        } else {
+            if (!inputStr.substring(6, 7).equals(" ")) {
+                throw new GrapieExceptions(ErrorMsg.noSpaceError);
+            } else {
+                String strNumberDeleted = inputStr.substring(7, inputStr.length());
+                strNumberDeleted.replaceAll("\\s+", ""); //remove all white spaces
+
+                boolean isANumber = isNumber(strNumberDeleted);
+
+                if (isANumber) {
+                    int numToDelete = Integer.parseInt(strNumberDeleted);
+
+                    if (storingList.size() >= numToDelete) {
+
+                        int newSize = storingList.size() - 1;
+                        String toPrint = " Alrighty. I've removed this task: \n"
+                                + storingList.get(numToDelete - 1)
+                                + "\n Now you have " + newSize + " tasks in the list.";
+
+                        storingList.remove(numToDelete - 1);
+                        storage.deleteLineFromHardDisk(numToDelete);
+                        return toPrint;
+                        //formattingDivider(toPrint);
+                        //delete from hard disk
+
+
+                    } else {
+                        throw new GrapieExceptions(ErrorMsg.numberDoNotExistError(numToDelete));
+                    }
+                } else {
+                    throw new GrapieExceptions(ErrorMsg.invalidNumberError);
+                }
+            }
+        }
```

Comment:
Same as completeTask. It might be best to have your inputs parsed beforehand and your logic in your main class.